src/components/OnboardingManager.tsx

```diff
@@ -0,0 +1,133 @@
+'use client';
+
+import {
+  BackLink,
+  FormDiv,
+  OuterDiv,
+  FormContainer,
+} from '@/app/onboarding/styles';
+import { P } from '@/styles/text';
+import { OnboardingContext } from '@/utils/OnboardingProvider';
+import { usePathname, useRouter } from 'next/navigation';
+import { ReactNode, useContext, useEffect, useMemo } from 'react';
+import ProgressBar from './ProgressBar';
+import BigButton from './BigButton';
+
+export default function OnboardingManager({
+  children,
+}: {
+  children: ReactNode;
+}) {
+  const onboarding = useContext(OnboardingContext);
+  const router = useRouter();
+  const pathname = usePathname();
+  const pageProgress = useMemo(() => {
+    if (!onboarding) return 0;
+    const find = onboarding.flow.findIndex(
+      f => `/onboarding/${f.url}` === pathname,
+    );
+    return find !== -1 ? find : 0;
+  }, [pathname, onboarding]);
+
+  useEffect(() => {
+    if (!onboarding) return;
+
+    // out of bounds redirect
+    if (
+      onboarding.progress < 0 ||
+      onboarding.progress >= onboarding.flow.length
+    ) {
+      router.push('/cases');
+      return;
+    }
+
+    // url not in flow
+    if (pageProgress === -1) {
+      router.replace(`/onboarding/${onboarding.flow[0].url}`);
+      onboarding.setProgress(0);
+      return;
+    }
+
+    // page visited before
+    if (pageProgress <= onboarding.progress) {
+      return;
+    }
+
+    // course correct
+    const correctURL = `/onboarding/${
+      onboarding.flow[onboarding.progress].url
+    }`;
+
+    if (pathname !== correctURL) {
+      router.replace(correctURL);
+    }
+  }, [onboarding, pathname, router, pageProgress]);
+
+  const advanceProgress = () => {
+    if (!onboarding) return;
+    const newProgress = pageProgress + 1;
+
+    if (newProgress > onboarding.progress) onboarding.setProgress(newProgress);
+
+    if (newProgress >= onboarding.flow.length) router.push('/cases');
+    else router.push(`/onboarding/${onboarding.flow[newProgress].url}`);
+  };
+
+  return (
+    <div>
+      <BackLink
+        href={
+          onboarding && pageProgress > 0
+            ? `/onboarding/${onboarding.flow[pageProgress - 1].url}`
+            : '/cases'
+        }
+      >
+        <P>Back</P>
+      </BackLink>
+      <OuterDiv>
+        <ProgressBar
+          steps={
+            new Set(onboarding ? onboarding.flow.slice(1).map(f => f.name) : [])
+          }
+          progress={pageProgress}
+        />
+        <FormContainer>
+          <FormDiv>
+            {children}
+            <BigButton
+              disabled={onboarding && onboarding.canContinue}
+              onClick={() => advanceProgress()}
+            >
+              <P $color="white">
+                {onboarding && pageProgress === onboarding.flow.length - 1
+                  ? 'Continue to Available Cases'
```

Comment:
curious why "Available Cases" goes to the next line here?

Reply:
"Continue to Available Cases" is the text displayed in the last onboarding screen

Reply:
LOOKS GOOD NOW!